inngest/src/main/kotlin/com/inngest/Comm.kt

```diff
@@ -166,8 +169,50 @@ class CommHandler(
     // TODO
 //    fun sync(): Result<InngestSyncResult> = Result.success(InngestSyncResult.None)
 
-    fun introspect(origin: String): String {
-        val requestPayload = getRegistrationRequestPayload(origin)
+    fun introspect(
+        signature: String?,
+        requestBody: String,
+        serverKind: String?,
+    ): String {
+        val insecureIntrospection =
+            InsecureIntrospection(
+                functionCount = functions.size,
+                hasEventKey = Environment.isInngestEventKeySet(client.eventKey),
+                hasSigningKey = config.hasSigningKey(),
+                mode = if (client.env == InngestEnv.Dev) "dev" else "cloud",
+            )
+
+        val requestPayload =
+            when (client.env) {
+                InngestEnv.Dev -> insecureIntrospection
+
+                else ->
+                    runCatching {
+                        checkHeadersAndValidateSignature(signature, requestBody, serverKind, config)
+
+                        SecureIntrospection(
+                            functionCount = functions.size,
+                            hasEventKey = Environment.isInngestEventKeySet(client.eventKey),
+                            hasSigningKey = config.hasSigningKey(),
+                            authenticationSucceeded = true,
+                            mode = "cloud",
+                            env = client.env.value,
+                            appId = config.appId(),
+                            apiOrigin = "${config.baseUrl()}/",
+                            framework = framework.value,
+                            sdkVersion = Version.getVersion(),
+                            sdkLanguage = "java",
```

Comment:
I went with java as the sdk language following this: https://linear.app/inngest/project/javakotlin-sdk-0e841a80efe2#projectUpdate-dd0137ce&comment-1288aae4
Let me know if we want a different value.